battlepass/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+			org_id: T::Hash,
+			battlepass_id: T::Hash,
+			nft_id: T::Hash,
+			block_number: T::BlockNumber
+		},
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		AuthorizationError,
+		MissingParameter,
+		InvalidParameter,
+		OrganizationUnknown,
+		OrgHasActiveBattlepass,
+		BattlepassExists,
+		BattlepassClaimed,
+		BattlepassUnknown,
+		NotMember,
+	  }
+
+	/// Battlepass by its id.
+	///
+	/// Battlepasses: map Hash => Battlepass
+	#[pallet::storage]
+	#[pallet::getter(fn get_battlepass)]
+	pub(super) type Battlepasses<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, Battlepass<T::Hash, T::AccountId, T::BlockNumber, String<T>>, OptionQuery>;
+
+	/// Number of battlepasses per organization.
+	///
+	/// BattlepassCount: map Hash => u32
+	#[pallet::storage]
+	#[pallet::getter(fn get_battlepass_count)]
+	pub type BattlepassCount<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, u32, ValueQuery>;
+
+	/// Current active battlepass in organization.
+	///
+	/// ActiveBattlepassByOrg: map Hash => Hash
+	#[pallet::storage]
+	#[pallet::getter(fn get_active_battlepass)]
+	pub type ActiveBattlepassByOrg<T: Config> = StorageMap<_, Blake2_128Concat, T::Hash, T::Hash, OptionQuery>;
+
+	/// Claimed Battlepass by user.
+	///
+	/// ClaimedBattlepass: map (AccountId, Hash) => Hash
+	#[pallet::storage]
+	#[pallet::getter(fn get_claimed_battlepass)]
+	pub(super) type ClaimedBattlepass<T: Config> = StorageDoubleMap<_,
+		Blake2_128Concat, T::AccountId,
+		Blake2_128Concat, T::Hash,
+		T::Hash,
+		OptionQuery
+	>;
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {
+		#[pallet::weight(0)]
+		pub fn create_battlepass(
+			origin: OriginFor<T>,
+			org_id: T::Hash,
+			name: String<T>,
+			cid: String<T>,
+			price: u16,
+		) -> DispatchResult {
+			let creator = ensure_signed(origin)?;
+
+			// check if origin is Org Prime or Root
+
+			// check if Org exists
+
+			// check if active battlepass does not exist for the Org
+
+
+			
+			// Create Battlepass
+			let now = <frame_system::Pallet<T>>::block_number();
+			let battlepass: Battlepass<T::Hash, T::AccountId, T::BlockNumber, String<T>> = types::Battlepass {
+				creator, 
+				org_id,
+				name,
+				cid,
+				state: types::BattlepassState::Active,
+				season: Self::get_battlepass_count(org_id),
+				price,
+				created: now.clone(), mutated: now
+			};
+			let battlepass_id = <T as frame_system::Config>::Hashing::hash_of(&battlepass);
+
+			Battlepasses::<T>::insert(&battlepass_id, battlepass);
+
+			Self::deposit_event(Event::BattlepassCreated { org_id, battlepass_id, block_number: now });
+
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		pub fn claim_battlepass(
+			origin: OriginFor<T>,
+			org_id: T::Hash,
+		) -> DispatchResult {
+			let claimer = ensure_signed(origin)?;
+			let battlepass_id = Self::get_active_battlepass(org_id).ok_or(Error::<T>::BattlepassUnknown)?;
+			let now = <frame_system::Pallet<T>>::block_number();
+
+			// check if user is a member of organization
+
+			// check if Org exists
```

Comment:
you can spare this lookup as the battlepass for an org which does not exist will also fail